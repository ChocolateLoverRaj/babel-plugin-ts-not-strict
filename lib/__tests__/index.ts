import plugin from '../'
import { transformSync } from '@babel/core'
import pluginSyntaxTypescript from '@babel/plugin-syntax-typescript'
import never from 'never'
import test, { Macro } from 'ava'

const macro: Macro<[string]> = (t, input): void => {
  const { code } = transformSync(input, {
    plugins: [pluginSyntaxTypescript, plugin]
  }) ?? never('No result')
  t.snapshot(code)
}

const arrowFn = '(a: string): string => a;'

/* test('export default with const', macro, `
const fn = ${arrowFn}

export default fn
`)

test('export default directly', macro, `
export default ${arrowFn}
`)

test('export named', macro, `
export const fn = ${arrowFn}
`)

test('function', macro, `
export default function add(a: number): number {
  return a
}
`)

test('function dts', macro, `
export function a(n: number): number;
`)

test('useControlledState', macro, `
export function useControlledState<T>(value: T, defaultValue: T, onChange: ` + `
(value: T, ...args: any[]) => void): [T, (value: T | ((prevState: T) => T), ...args: any[]) => void];
`)

test('function parameter', macro, `
export function fn(cb: () => void): void;
`)

test('already parenthesized function parameter', macro, `
export function fn(cb: (() => void)): void;
`)

test('already includes undefined', macro, `
export function fn(a: string | undefined): void;
`) */

/* test('already optional', macro, `
export function fn(a?: string): void;
`) */

test('interface', macro, `
export interface Person {
  name: string
}
`);

test('interface param', macro, `
interface Person {
  name: string
}
export function fn(a: Person): void;
`)
