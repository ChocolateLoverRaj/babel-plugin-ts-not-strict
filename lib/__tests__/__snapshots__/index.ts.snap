// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`already includes undefined 1`] = `"export function fn(a: string | undefined): void;"`;

exports[`already optional 1`] = `"export function fn(a?: string): void;"`;

exports[`already parenthesized function parameter 1`] = `"export function fn(cb: (() => void) | undefined): void;"`;

exports[`export default directly 1`] = `"export default ((a: string | undefined): string => a);"`;

exports[`export default with const 1`] = `
"const fn = (a: string | undefined): string => a;

export default fn;"
`;

exports[`export named 1`] = `"export const fn = (a: string | undefined): string => a;"`;

exports[`function 1`] = `
"export default function add(a: number | undefined): number {
  return a;
}"
`;

exports[`function dts 1`] = `"export function a(n: number | undefined): number;"`;

exports[`function parameter 1`] = `"export function fn(cb: (() => void) | undefined): void;"`;

exports[`interface 1`] = `
"export interface Person {
  name?: string;
}"
`;

exports[`useControlledState 1`] = `"export function useControlledState<T>(value: T | undefined, defaultValue: T | undefined, onChange: ((value: T, ...args: any[]) => void) | undefined): [T, (value: T | ((prevState: T) => T), ...args: any[]) => void];"`;
